src/taco-livereload/lib/utils/homePage/homePage.html

```diff
@@ -0,0 +1,103 @@
+        </style>
+        <script src="js/q/1.0.1/q.js"></script>
+        <script src="js/underscore/1.7.0/underscore-min.js"></script>
+    </head>
+    
+    <body>
+        <h3>CORDOVA-PLUGIN-LIVERELOAD</h3>
+        <div id="error">There was an error connecting to the BrowserSync server.
+            <br />
+            <br />Try using the<b><a>'tunnel'</a></b>option as in:
+            <br />
+            <b>`cordova run android -- --livereload --tunnel`</b>
+            <br />
+            <br />For more information,
+            <b>
+                <a href="https://github.com/omefire/cordova-plugin-livereload#options">check the documentation</a>
+            </b>
+            <br />
+            <br />
+        </div>
+        <div id="loading">...Loading your app, Please wait...</div>
+    </body>
+    <script type="text/javascript">
+        
+        /**
+         * Make an AJAX request:
+         *   - url: the url to make the AJAX request to
+         *   - success: the function to call in case the AJAX call succeeds
+         *   - fail: the function to call in case the AJAX call fails
+         */
+        function ajaxCall(url, success, fail) {
+            var req = new XMLHttpRequest();			
+        			
+            // Use `Math.random()` to prevent the caching of the server_url by the browser
+            req.open('GET', url + '?' + Math.random());        			
+            req.onreadystatechange = function(aEvt) {
+                if (req.readyState == 4) {
+                    if (req.status == 200) {
+                        return success();
+                    }
+                    return fail();
+                }
+            };
+            req.send();
+        }
+        
+        /**
+         * Checks if a particular server can be connected from this page
+         */
+        function isServerAccessible(server_url) {
+            if(!server_url) {
+                return Q(false);
+            }
+                        
+            var deferred = Q.defer();            
+            ajaxCall(
+                server_url, 
+                function() {
+                    deferred.resolve(true);
+                }, 
+                function() {
+                    deferred.resolve(false);
+                }
+            );            
+            return deferred.promise;
+        }
+        		
+        // Note: __SERVERS__ gets replaced by real server urls right before this file gets deployed to a device/emulator
+        var SERVERS = JSON.parse('__SERVERS__');
+        
+        // Note: These are the kinds of servers we are interested in. (e.g: we don't care about 'ui' and 'ui-external')
+        var serverTypes = ['local', 'external', 'tunnel'];
+        
+        var areServersAccessiblePromises = serverTypes.map(function(serverType) {
+            return isServerAccessible(SERVERS[serverType]);
+        });
+        
+        Q.all(areServersAccessiblePromises).then(function(areServersAccessible) {
```

Comment:
I don't believe we can. The code appears to build an array of booleans indicating if we can ping a sever. We then check those responses in order of preference. Does Q.any guarantee ordering of execution?